### PR TITLE
feat: allow tex to pdf implementation via external service

### PR DESCRIPTION
**Pull request by mussbach**
This PR adds an alternative way to create pdf exports via an external api service. The existing implementation via rabbitmq is still the default one, it is possible to switch via Interface implementation.

### How to review/test
code review might be best. If you have access to the new service you could use it.

### PR Checklist

Delete the checkbox if it doesn't apply/isn't necessary.

- [ ] Tests updated/created
- [ ] Update documentation
- [ ] Link all relevant tickets
- [ ] Move the tickets on the board accordingly

### CODE REVIEW

File: client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMetaAttachments.vue
Comment (graupnerdemos) [inline client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMetaAttachments.vue:1]: Is this meant to be changed in here?

File: demosplan/DemosPlanCoreBundle/DoctrineMigrations/2023/06/Version20230601095828.php
Comment (graupnerdemos) [inline demosplan/DemosPlanCoreBundle/DoctrineMigrations/2023/06/Version20230601095828.php:1]: This also doesn't seem like it belongs here.

File: demosplan/DemosPlanCoreBundle/Controller/User/DemosPlanUserController.php
Comment (graupnerdemos) [inline demosplan/DemosPlanCoreBundle/Controller/User/DemosPlanUserController.php:1]: not related to pdf stuff

File: demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanAssessmentController.php
Comment (graupnerdemos) [inline demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanAssessmentController.php:1]: this seems to be from another change

**Review by graupnerdemos — CHANGES_REQUESTED** [review]
This PR contains a bunch of changes which are completely unrelated to PDF handling and seem to have barged in via merges. @mussbach can you please check that?

**Comment by mussbach** [comment]
@graupnerdemos thx, fixed

File: demosplan/DemosPlanCoreBundle/Tools/PdfCreatorRabbitmq.php [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                'Fehler in ImportConsumer:',
+                [$e]
+            );
+            throw new TimeoutException('Timeout ');
+        } catch (Exception $e) {
+            $this->logger->error(
+                'Could not create PDF ',
+                [$e]
+            );
+            throw new Exception('Could not create PDF ');
```
Comment (graupnerdemos) [inline demosplan/DemosPlanCoreBundle/Tools/PdfCreatorRabbitmq.php:109]: Just once here: I'd prefer having a PdfCreateException and static constructors for the different cases, but I am unsure whether this is part of our coding guidelines or just something I've seen and done a bunch of times.

**Review by graupnerdemos — APPROVED** [review]
One minor remark. However, since that is mostly just moved code from the previous solution I don't deem it blocking for this change.

File: demosplan/DemosPlanCoreBundle/Tools/PdfCreatorRabbitmq.php (hunk 1 above)
Comment (mussbach) [reply demosplan/DemosPlanCoreBundle/Tools/PdfCreatorRabbitmq.php:109]: I am quite sure that this is currently not part of our coding guidelines as this has never been discussed (at least I am not aware of it and I dislike the static constructor). But having specialized Exceptions is better

**Pull request merged**